test/system/application-scans/node.spec.ts

```diff
@@ -216,21 +217,18 @@ describe("node application scans", () => {
     expect(depGraphNpmFromGlobalNodeModules.rootPkg.name).toEqual("lib");
   });
 
-  it("should extract image content successfully", async () => {
+  it("should collect application files successfully", async () => {
     const fixturePath = getFixture("npm/multi-project-image.tar");
     const imageNameAndTag = `docker-archive:${fixturePath}`;
-    const result = await extractContent([getNodeAppFileContentAction], {
+    const result = await collectApplicationFiles({
       path: imageNameAndTag,
     });
-    expect(Object.keys(result.extractedLayers).length).toEqual(608);
-    Object.keys(result.extractedLayers).forEach((fileName) => {
-      expect(
-        fileName.endsWith("/package.json") ||
-          fileName.endsWith("/package-lock.json") ||
-          fileName.endsWith("/yarn.lock"),
-      ).toBeTruthy();
-      expect("node-app-files" in result.extractedLayers[fileName]).toBeTruthy();
-    });
+    const applicationFilesFact: ApplicationFilesFact = result.scanResults[0].facts.find(
+        (fact) => fact.type === "applicationFiles",
+      )!.data;
+
+    const fileList = applicationFilesFact.data[0].fileHierarchy.map((fh) => fh.path);
```

Comment:
Why map if you're only counting?

Reply:
this is the model:
```
export interface ApplicationFileInfo {
  path: string;
}
export interface ApplicationFiles {
  fileHierarchy: ApplicationFileInfo[];
  moduleName?: string;
  language: string;
}
```
so in order to pull out the list of files - i have to extract each path by its own...